### ISSUE TITLE
Simulation not periodic, while it should be

### ISSUE THREAD

**Issue by arionaris** [PHAREHUB/PHARE#689]
The following python code : 

```python 
#!/usr/bin/env python3

import pyphare.pharein as ph #lgtm [py/import-and-import-from]
from pyphare.pharein import Simulation
from pyphare.pharein import MaxwellianFluidModel
from pyphare.pharein import ElectromagDiagnostics,FluidDiagnostics, ParticleDiagnostics
from pyphare.pharein import ElectronModel
from pyphare.simulator.simulator import Simulator, startMPI

from pyphare.pharein import global_vars as gv

import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl
mpl.use('Agg')

#from pyphare.cpp import cpp_lib
#cpp = cpp_lib()
#startMPI()

diag_outputs="/mnt/beegfs/workdir/nathan.guyot/run_043/diag"
out="/mnt/beegfs/workdir/nathan.guyot/run_043/restart/"

from datetime import datetime



def density(x, y):
    #L = sim.simulation_domain()[1]
    return 1.


def S(y, y0, l):
    return 0.5*(1. + np.tanh((y-y0)/(l)))


def by(x, y):
    return 0.


def bx(x, y):
    return 0.


def bz(x, y):
    return 1.


def b2(x, y):
    return bx(x,y)**2 + by(x, y)**2 + bz(x, y)**2


def T(x, y):
    K = 1
    temp = 1./density(x, y)*(K - b2(x, y)*0.5)
    assert np.all(temp >0)
    return temp

def vx(x, y):
    from pyphare.pharein.global_vars import sim
    y0 = gv.sim.simulation_domain()[1]
    x_b = gv.sim.simulation_domain()[0]
    amp = 0.1
    k = 4
    x0 = x_b*0.25
    x1 = x_b*0.75
    v0 = amp*np.sin(((2*np.pi)/y0)*y*k)*np.exp(-(((x-x0)**2)/2))
    v1 = amp*np.sin(((2*np.pi)/y0)*y*k)*np.exp(-(((x-x1)**2)/2))
    return(v0 + v1)

def vy(x, y):
    from pyphare.pharein.global_vars import sim
    x0 = gv.sim.simulation_domain()[0]
    v1 = -0.5
    v2 = 0.5
    return(v1 + (v2-v1)*(S(x,x0*0.25,0.5) - S(x,x0*0.75,0.5)))
        

def vz(x, y):
    return 0.
    

def vthx(x, y):
    return 0.2
    

def vthy(x, y):
    return 0.2
    
    
def vthz(x, y):
    return 0.2


def config():

    Simulation(
        time_step_nbr=20000,
        time_step=0.005,
        #boundary_types="periodic",
        cells=(200,100),
        dl=(0.4, 0.4),
        refinement="tagging",
        max_nbr_levels = 3,
        tag_buffer = 8,
        #smallest_patch_size = 20,
        nesting_buffer=1,
        refinement_boxes={},
        hyper_resistivity=0.005,
        resistivity=0.001,
        diag_options={"format": "phareh5",
                      "options": {"dir": diag_outputs,
                                  "mode":"overwrite"}},
        strict=True,
    )


    vvv = {
        "vbulkx": vx, "vbulky": vy, "vbulkz": vz,
        "vthx": vthx, "vthy": vthy, "vthz": vthz,
        "nbr_part_per_cell":100
    }

    MaxwellianFluidModel(
        bx=bx, by=by, bz=bz,
        protons={"charge": 1, "density": density,  **vvv},

    )

    ElectronModel(closure="isothermal", Te=0.0)

    sim = ph.global_vars.sim
    dt = 100*sim.time_step
    nt = sim.final_time/dt+1
    timestamps = (dt * np.arange(nt))
    

    for quantity in ["E", "B"]:
        ElectromagDiagnostics(
            quantity=quantity,
            write_timestamps=timestamps,
            compute_timestamps=timestamps,
        )
        
    for quantity in ["density", "bulkVelocity"]:
        FluidDiagnostics(
            quantity=quantity,
            write_timestamps=timestamps,
            compute_timestamps=timestamps,
            )


    for name in ["domain","levelGhost", "patchGhost"]:
        ParticleDiagnostics(quantity=name,
                            compute_timestamps=timestamps,
                            write_timestamps=np.array([sim.time_step*a*2000 for a in range(0,11)]),
                            population_name="protons")


def get_time(path, time, datahier = None):
    time = "{:.10f}".format(time)
    from pyphare.pharesee.hierarchy import hierarchy_from
    datahier = hierarchy_from(h5_filename=path+"/EM_E.h5", time=time, hier=datahier)
    datahier = hierarchy_from(h5_filename=path+"/EM_B.h5", time=time, hier=datahier)
    return datahier

def post_advance(new_time):
    if cpp.mpi_rank() == 0:
        print(f"running tests at time {new_time}")
        from tests.simulator.test_advance import AdvanceTestBase
        test = AdvanceTestBase()
        test.base_test_overlaped_fields_are_equal(get_time(diag_outputs, new_time), new_time)
        print(f"tests passed")

def main():
    config()
    s = Simulator(gv.sim)
    s.initialize()
    s.run()


if __name__=="__main__":
    main()


```

Runs a simulation with perturbed ```vx(x,y)```

Which raises the following :  ```RuntimeError: Simulation is not periodic```

The figure : 

![Figure 3(3)](https://user-images.githubusercontent.com/53024996/178230291-60a9f099-7fad-489b-b72c-352d5a302f35.png)

shows the simulation is in fact periodic, border wise, and should not raise the RuntimeError.

This happens, with ```cells=(200,100)``` in  ```Simulation```

However, upong taking ```cells=(200,200)``` , the RuntimeError does not occur, and the simulation is considered periodic, while both cases verify the criteria of border values as far as $2e^{-15}$.

**Comment by PhilipDeegan** [comment]
should be fixed in https://github.com/PHAREHUB/PHARE/pull/698

**Comment by nicolasaunai** [comment]
closed after #703
